File: Leetcode/405. Convert a Number to Hexadecimal/convert_a_number_to_hexadecimal.py
class Solution:
    def toHex(self, num: int) -> str:
        # handling zero case
        if num==0:
            return "0"
        # 2's complement
        if num<0:
            print(num & (2**32-1),(1<<32)+num)
            # num = (1<<32)+num
            num &= (2**32-1)
        hex_str ="0123456789abcdef"
        hex_digits=""
        while num>0:
            digit = num%16
            hex_digits = hex_str[digit] +hex_digits
            num//=16
        return hex_digits

sol = Solution()
print(sol.toHex(-1))

File: Leetcode/405. Convert a Number to Hexadecimal/convert_a_number_to_hexadecimal.rs
struct Solution{}

impl Solution {
    pub fn to_hex(num: i32) -> String {
        let mut num = num as u32;
        if num==0{
            return '0'.to_string()
        }
        let hex_str  =["0","1","2","3","4","5","6","7","8","9","a","b","c","d","e","f"];
        let mut hex_code ="".to_string();
        while num>0{
            let digit = num%16;
            hex_code.insert_str(0,hex_str[digit as usize]);
            num>>=4;
        }
        hex_code
    }
}
fn main() {
    let res = Solution::to_hex(26);
    println!("{}",res);
}
